frontend/src/metabase/components/OnClickOutsideWrapper.jsx

```diff
@@ -39,11 +39,14 @@ export default class OnClickOutsideWrapper extends Component {
         window.removeEventListener("click", this._handleClick, true);
         clearTimeout(this._timeout);
 
-        // remove popover from the stack
-        var index = popoverStack.indexOf(this);
-        if (index >= 0) {
-            popoverStack.splice(index, 1);
-        }
+        // remove from the stack after a delay, if it is removed through some other
+        // means this will happen too early causing parent modal to close
+        setTimeout(() => {
+            var index = popoverStack.indexOf(this);
+            if (index >= 0) {
+                popoverStack.splice(index, 1);
```

Comment:
Saw this code just now. It's great that it works, but we might want to add a TODO to eventually consider refactoring `popoverStack` into a prop passed down from a reducer somewhere. It just seems really hacky to have a component depend on a random floating variable that doesn't even live in any component internal state, let alone in any reducers. Feels like this could become problematic if we want to make use of some of the nicer tooling available for redux later down the line.